File: julia/2017/1/solution1.jl
f = open("../../../data/2017/1/data.txt")
captcha = readstring(f)
close(f)
captcha = replace(string(captcha, captcha[1]), "\n", "")

total = 0
for i = 1:length(captcha)-1
    if captcha[i] == captcha[i+1]
        total += parse(Int, captcha[i])
    end
end

println("Solution is $total")

File: python/2017/1/solution1.py
with open('./data.txt', 'rb') as f:
    captcha = f.read().strip()
captcha += captcha[0]

total = 0
for i, digit in enumerate(captcha):
    try:
        if digit == captcha[i+1]:
            total += int(digit)
    except IndexError:
        pass

print "Solution is %d" % total